packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts

```diff
@@ -0,0 +1,207 @@
+/*!
+ * Copyright (c) 2015-2020 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@webex/test-helper-chai';
+import FakeTimers from '@sinonjs/fake-timers';
+import Metrics, {config} from '@webex/internal-plugin-metrics';
+import MockWebex from '@webex/test-helper-mock-webex';
+import sinon from 'sinon';
+import {WebexHttpError} from '@webex/webex-core';
+
+function promiseTick(count) {
+  let promise = Promise.resolve();
+
+  while (count > 1) {
+    promise = promise.then(() => promiseTick(1));
+    count -= 1;
+  }
+
+  return promise;
+}
+
+describe('plugin-metrics', () => {
+  describe('CallDiagnosticEventsBatcher', () => {
+    let webex;
+
+    beforeEach(() => {
+      //@ts-ignore
+      webex = new MockWebex({
+        children: {
+          metrics: Metrics,
+        },
+      });
+
+      webex.config.metrics = config.metrics;
+
+      webex.request = function (options) {
+        return Promise.resolve({
+          statusCode: 204,
+          body: undefined,
+          options,
+        });
+      };
+      sinon.spy(webex, 'request');
```

Comment:
nitpick: you could just do this in 1 line:
```
webex.request = sinon.stub().resolves({
  statusCode: 204,
  body: undefined,
  options
});
```

Reply:
this was copy pasted from old batcher tests which are still working, so will leave it as is for now.